### ISSUE TITLE
Is YOLOV5 suitable for self-training face models identified?

### ISSUE THREAD

**Issue by Kingstom8MB** [ultralytics/yolov5#2035]
## ❔Question

Can Yolov5 be used for identification?I have tried to train my own face data, but the error in video recognition is very large, and even the recognition in the pictures used for training can not get the ideal result, I wonder why

**Comment by Kingstom8MB** [comment]
I detect a face in a video and I capture it and I save it
This is the picture that was saved.It has 424 copies
![image](https://user-images.githubusercontent.com/50566074/105652896-6639e200-5ef5-11eb-98c7-da4f3ad2053d.png)
The target image is intercepted in advance, and then the script is used to generate the corresponding TXT annotation of the full image in bulk
![image](https://user-images.githubusercontent.com/50566074/105653122-eeb88280-5ef5-11eb-9cc8-153c50eadced.png)
This is the contents of the generated file
![image](https://user-images.githubusercontent.com/50566074/105653186-1c053080-5ef6-11eb-8058-28c4ff7baeb5.png)
Then edit the YALM file label and training path
![image](https://user-images.githubusercontent.com/50566074/105653288-625a8f80-5ef6-11eb-9e61-f11920143acb.png)
The command was used for training. Due to the concern that the model was not accurate enough, the epoch parameters were increased：python3.9 train.py --img 640 --batch 16 --epochs 1000 --data ../mask/data.yaml --weights yolov5s.pt
![image](https://user-images.githubusercontent.com/50566074/105653529-ffb5c380-5ef6-11eb-83fa-c1c0675c2ae8.png)
He then used the generated best.pt model to detect it, but instead of a face, he detected other scenes with very low confidence
The last video that I used for detection is the same video that I used for face detection

**Comment by Kingstom8MB** [comment]
![image](https://user-images.githubusercontent.com/50566074/105664248-28e24e00-5f0f-11eb-863e-d06ba4ac6505.png)

This is the result of identification, it's very inaccurate, I don't know what went wrong

**Comment by decent-engineer-decent-datascientist** [comment]
Your training images should look like your testing images. Aka, a full frame with labels generated from something like labelimg. Unless I'm missing something, it seems you're feeding the training process crops of just the face?

**Comment by glenn-jocher** [comment]
@Kingstom8MB this repo's job is to train on your data to a high mAP on your val data. This has been more than achieved with a 1.0 mAP from your screenshot.

It's up to you as the ML engineer to structure your dataset correctly to generalize to your domain. We can't do that for you.
<img width="2029" alt="Screen Shot 2021-01-07 at 9 18 07 PM" src="https://user-images.githubusercontent.com/26833433/105757864-69d46580-5f03-11eb-9b93-9c2a84b503f3.png">

**Comment by Kingstom8MB** [comment]
> Your training images should look like your testing images. Aka, a full frame with labels generated from something like labelimg. Unless I'm missing something, it seems you're feeding the training process crops of just the face?

Yeah, I wanted it to be able to identify people by face recognition, so I trained my face data first. Take a test. But it seems to have failed this time. Maybe the data source is not good enough. Maybe I should take life photos with higher resolution for training and use them for surveillance video recognition?

**Comment by Kingstom8MB** [comment]
> @Kingstom8MB this repo's job is to train on your data to a high mAP on your val data. This has been more than achieved with a 1.0 mAP from your screenshot.
> 
> It's up to you as the ML engineer to structure your dataset correctly to generalize to your domain. We can't do that for you.
> <img width="2029" alt="Screen Shot 2021-01-07 at 9 18 07 PM" src="https://user-images.githubusercontent.com/26833433/105757864-69d46580-5f03-11eb-9b93-9c2a84b503f3.png">
> 
> 
> 
> 

Thank you for your answer. Although I didn't quite understand it, I am just coming into contact with machine AI and I am not familiar with it. I will study it again

**Comment by decent-engineer-decent-datascientist** [comment]
@Kingstom8MB you were on the right track, the only mistake you made was in the generation of the training data. You need to label images as shown by the yellow box:
![105664248-28e24e00-5f0f-11eb-863e-d06ba4ac6505](https://user-images.githubusercontent.com/77806775/105924780-a20f9d00-6004-11eb-9de4-44dc2e48e1df.png)



Use this repo to label your dataset, make sure to change format to YOLO:
https://github.com/tzutalin/labelImg

**Comment by Kingstom8MB** [comment]
> @Kingstom8MB you were on the right track, the only mistake you made was in the generation of the training data. You need to label images as shown by the yellow box:
> ![105664248-28e24e00-5f0f-11eb-863e-d06ba4ac6505](https://user-images.githubusercontent.com/77806775/105924780-a20f9d00-6004-11eb-9de4-44dc2e48e1df.png)
> 
> 
> 
> Use this repo to label your dataset, make sure to change format to YOLO:
> https://github.com/tzutalin/labelImg
> 

Thanks for your help. I will try to re-label and make the data set in the near future, and check the performance effect later

**Comment by Kingstom8MB** [comment]
Maybe my previous method was wrong, instead of using the script to automatically capture the face image, I should use such manual marking method. Anyway, I will try it and then reply the effect. Thank you for your help

**Comment by glenn-jocher** [comment]
OIv6 already has face as a detection category, you can train YOLOv5 on this dataset with a subset of the images that contain your category of interest. Other organizations have done this successfully.
https://storage.googleapis.com/openimages/web/index.html

**Comment by Kingstom8MB** [comment]
> OIv6已经将人脸作为检测类别，您可以使用包含您感兴趣的类别的图像子集在此数据集上训练YOLOv5。其他组织已成功完成此操作。
> https://storage.googleapis.com/openimages/web/index.html

It is a good suggestion, I now use yolov5 track, on the basis of using PyFaceDet confirm whether for face recognition, due to my computer performance is very poor, but my goal is to make a real-time system, so now I am trying to optimize its performance, I'll try this method and before deciding whether to replace the face detection scheme, thank you very much.
